src/couch_dist/src/couch_dist.erl

```diff
@@ -0,0 +1,114 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_dist).
+
+-export([childspecs/0, listen/2, accept/1, accept_connection/5,
+    setup/5, close/1, select/1, is_node_name/1]).
+
+% Just for tests
+-export([no_tls/1, get_init_args/0]).
+
+
+childspecs() ->
+    {ok, [{ssl_dist_sup, {ssl_dist_sup, start_link, []},
+        permanent, infinity, supervisor, [ssl_dist_sup]}]}.
+
+
+listen(Name, Host) ->
+    NodeName =
+        case is_atom(Name) of
+            true -> atom_to_list(Name);
+            false -> Name
+        end,
+    case no_tls(NodeName ++ "@" ++ Host) of
+        true -> inet_tcp_dist:listen(NodeName, Host);
+        false -> inet_tls_dist:listen(NodeName, Host)
+    end.
+
+
+accept(Listen) ->
+    case no_tls(node()) of
+        true -> inet_tcp_dist:accept(Listen);
+        false -> inet_tls_dist:accept(Listen)
+    end.
+
+
+accept_connection(AcceptPid, DistCtrl, MyNode, Allowed, SetupTime) ->
+    case no_tls(MyNode) of
+        true -> inet_tcp_dist:accept_connection(
+            AcceptPid, DistCtrl, MyNode, Allowed, SetupTime);
+        false -> inet_tls_dist:accept_connection(
+            AcceptPid, DistCtrl, MyNode, Allowed, SetupTime)
+    end.
```

Comment:
It took me a while to figure out how those two clauses were different. Would it be more clear to write like this?
```erlang
    Mod = case no_tls(MyNode) of
        true -> inet_tcp_dist;
        false -> inet_tls_dist
    end,
    Mod:inet_tcp_dist(AcceptPid, DistCtrl, MyNode, Allowed, SetupTime).
```
?

Maybe even factor that out as a function and use it in `accept_connection` and `setup`?
```erlang
inet_dist(Node) ->
    case no_tls(Node) of
        true -> inet_tcp_dist;
        false -> inet_tls_dist
    end.
```

Reply:
Good idea, modified, thanks for your comment.